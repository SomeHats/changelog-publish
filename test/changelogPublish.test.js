// @flow
const path = require("path");
const fs = require("fs");
const { promisify } = require("util");
const changelogPublish = require("../lib/changelogPublish");

const readFileAsync = promisify(fs.readFile);

describe("changelogPublish", () => {
  const testFixture = async name => {
    it(`fixture: ${name}`, async () => {
      const fixturePath = path.join(__dirname, "fixtures", name);
      const fixture = await readFileAsync(fixturePath, "utf-8");

      expect(
        changelogPublish(fixture, "1.2.3", new Date("2019-06-20"))
      ).toMatchSnapshot();
    });
  };

  testFixture("empty.md");
  testFixture("no-title.md");
  testFixture("no-section-unreleased.md");
  testFixture("no-versions.md");
  testFixture("valid.md");
});
