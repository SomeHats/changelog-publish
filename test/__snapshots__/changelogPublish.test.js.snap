// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changelogPublish fixture: empty.md 1`] = `
Object {
  "isValid": false,
  "message": "Changelog must have an 'Unreleased' section",
}
`;

exports[`changelogPublish fixture: no-section-unreleased.md 1`] = `
Object {
  "isValid": false,
  "message": "Changelog must have an 'Unreleased' section",
}
`;

exports[`changelogPublish fixture: no-title.md 1`] = `
Object {
  "changelog": "This is a changelog!

## [Unreleased]

## [1.2.3] - 2019-06-20

## [1.0.0] - 2019-06-19

### Added

-   Something
",
  "isValid": true,
}
`;

exports[`changelogPublish fixture: no-versions.md 1`] = `
Object {
  "changelog": "# Changelog

This is a changelog!

## [Unreleased]

## [1.2.3] - 2019-06-20

-   Something
",
  "isValid": true,
}
`;

exports[`changelogPublish fixture: valid.md 1`] = `
Object {
  "changelog": "# Changelog

This is a changelog!

## [Unreleased]

## [1.2.3] - 2019-06-20

-   Something

## [1.2.0] - 2019-06-19

-   Something else
",
  "isValid": true,
}
`;
